### PR TITLE
Subplots

### PR DESCRIPTION
**Pull request by kiamesdavies**
Thanks for contributing.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://developercertificate.org)

## Description

Added support for subplots as demo in the JavaScript API [here](https://plot.ly/javascript/subplots)

## Testing

A disabled test with an example on how to use the grid was added

### CODE REVIEW

File: core/src/test/java/tech/tablesaw/api/TableTest.java [hunk 1]
```diff
@@ -214,7 +214,7 @@ public void testSampleSplit() throws Exception {
         Table[] results = t.sampleSplit(.75);
         assertEquals(t.rowCount(), results[0].rowCount() + results[1].rowCount());
     }
-
+
```
Comment (benmccann) [inline core/src/test/java/tech/tablesaw/api/TableTest.java]: Please remove this extra added whitespace

File: jsplot/src/test/java/tech/tablesaw/components/GridTest.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    @Test
+    public void asJavascript() {
+        Grid x = Grid.builder()
+                .rows(10)
+                .columns(5)
+                .rowOrder(Grid.RowOrder.BOTTOM_TO_TOP)
+                .pattern(Grid.Pattern.INDEPENDENT)
+                .build();
+
+        System.out.println(x);
```
Comment (benmccann) [inline jsplot/src/test/java/tech/tablesaw/components/GridTest.java]: It'd be nicer to make an assertion here than printing to standard out

File: jsplot/src/main/java/tech/tablesaw/plotly/traces/AbstractTrace.java [hunk 3]
```diff
@@ -121,4 +145,6 @@ public HoverLabel hoverLabel() {
     public boolean showLegend() {
         return showLegend;
     }
+
```
Comment (lwhite1) [inline jsplot/src/main/java/tech/tablesaw/plotly/traces/AbstractTrace.java]: We don't need the added whitespace here.

File: jsplot/src/main/java/tech/tablesaw/plotly/traces/HistogramTrace.java [hunk 4]
```diff
@@ -40,8 +38,7 @@ private HistogramTrace(HistogramBuilder builder) {
         this.autoBinY = builder.autoBinY;
         this.opacity = builder.opacity;
         this.marker = builder.marker;
-        this.xAxis = builder.xAxis;
-        this.yAxis = builder.yAxis;
+
```
Comment (lwhite1) [inline jsplot/src/main/java/tech/tablesaw/plotly/traces/HistogramTrace.java]: extra line

File: jsplot/src/test/java/tech/tablesaw/components/GridTest.java [hunk 5]
```diff
@@ -16,6 +16,13 @@ public void asJavascript() {
...
                 .build();
 
-        System.out.println(x);
+        String asJavascript = x.asJavascript();
+        assertTrue(asJavascript.contains("rows"));
+        assertTrue(asJavascript.contains("columns"));
+        assertTrue(asJavascript.contains("roworder"));
+        assertTrue(asJavascript.contains("pattern"));
     }
+
```
Comment (lwhite1) [inline jsplot/src/test/java/tech/tablesaw/components/GridTest.java]: lines

**Comment by lwhite1** [comment]
This will be a huge improvement to the plotting capabilities. If you could just cleanup the extra blank lines added in various places it would be good to go.

**Comment by lwhite1** [comment]
@kiamesdavies  Thanks very much.

**Comment by kiamesdavies** [comment]
> @kiamesdavies Thanks very much.
You welcome

**Pull request merged**